contracts/LossOnFeeChecker.sol

```diff
@@ -0,0 +1,87 @@
+pragma solidity 0.6.12;
+pragma experimental ABIEncoderV2;
+
+interface IVault {
+    struct StrategyParams {
+        uint performanceFee;
+        uint activation;
+        uint debtRatio;
+        uint rateLimit;
+        uint lastReport;
+        uint totalDebt;
+        uint totalGain;
+        uint totalLoss;
+    }
+    function totalAssets() external view returns (uint);
+    function managementFee() external view returns (uint);
+    function performanceFee() external view returns (uint);
+    function strategies(address) external view returns (StrategyParams memory);
+    function lastReport() external view returns (uint);
+}
+
+interface IStrategy {
+    function vault() external view returns (address);
+}
+
+/// @title LossOnFeeChecker
+/// @notice Designed to prevent Management fees from creating lossy reports on Yearn vaults with API < 0.3.5
+/// @dev Begining with vaults API v0.3.5 management fees are adjust dynamically on report to prevent loss
+contract LossOnFeeChecker {
+
+    uint constant MAX_BPS = 10_000;
+    uint constant SECS_PER_YEAR = 31_557_600;
+
+    /// @notice Check if harvest does not contain a loss after fees
+    /// @dev should be called automically report transaction
+    function checkLoss(uint gain, uint loss) external view returns (uint) {
+        return _check(msg.sender, gain, loss);
+    }
+
+    /// @notice For testing amounts and strategies not directly on chain
+    function checkLoss(address strategy, uint gain, uint loss) external view returns (uint) {
```

Comment:
meant to support off-chain checks, so that for example the multisig can estimate the losses we expect need to be made-up to get the strategy harvesting again successfully.